src/components/Farmhand/Farmhand.js

```diff
@@ -537,69 +572,7 @@ export default class Farmhand extends Component {
 
     Object.assign(this.keyHandlers, {
       clearPersistedData: () => this.clearPersistedData(),
-      waterAllPlots: () => this.waterAllPlots(this.state),
-    })
-  }
-
-  initReducers() {
-    ;[
-      'addCowToInventory',
-      'addPeer',
-      'adjustLoan',
-      'changeCowAutomaticHugState',
-      'changeCowBreedingPenResident',
-      'changeCowName',
-      'clearPlot',
-      'computeStateForNextDay',
-      'fertilizePlot',
-      'forRange',
-      'harvestPlot',
-      'hugCow',
-      'makeRecipe',
-      'makeFermentationRecipe',
-      'modifyCow',
-      'offerCow',
-      'plantInPlot',
-      'prependPendingPeerMessage',
-      'purchaseCombine',
-      'purchaseComposter',
-      'purchaseCow',
-      'purchaseCowPen',
-      'purchaseCellar',
-      'purchaseField',
-      'purchaseItem',
-      'purchaseSmelter',
-      'purchaseStorageExpansion',
-      'removeCowFromInventory',
-      'removeKegFromCellar',
-      'removePeer',
-      'selectCow',
-      'sellCow',
-      'sellItem',
-      'sellKeg',
-      'setScarecrow',
-      'setSprinkler',
-      'showNotification',
-      'updatePeer',
-      'upgradeTool',
-      'waterAllPlots',
-      'waterField',
-      'waterPlot',
-      'withdrawCow',
-    ].forEach(reducerName => {
-      const reducer = reducers[reducerName]
-
-      if (process.env.NODE_ENV === 'development') {
-        if (typeof reducer === 'undefined') {
-          throw new Error(
-            `Reducer ${reducerName} is not exported from reducers/index.js`
-          )
-        }
-      }
-
-      this[reducerName] = (...args) => {
-        this.setState(state => reducer(state, ...args))
-      }
+      waterAllPlots: () => this.waterAllPlots(),
```

Comment:
can this just be `waterAllPlots: this.waterAllPlots`? or is there a scope issue if you do that?

Reply:
It can, but I'd prefer to keep this as-is for clarity's sake. It would be a harmless change in this case, but this change would obfuscate any function parameters (even though there are none in this case):

```suggestion
      waterAllPlots: this.waterAllPlots,
```

Tangentially, this is a change I would have likely made in the past. However, I've found that clever micro-optimizations like this tend to make code harder to follow. There's still plenty of them throughout this code base, but I am gradually trying to make Farmhand's code more obvious and straightforward by allowing for slightly more verbosity. 🙂